### ISSUE TITLE
App crashes in Value::toString() when trying to hot reload (iOS)

### ISSUE THREAD

**Issue by sryze** [facebook/react-native#22233]
## Environment

```
  React Native Environment Info:
    System:
      OS: macOS High Sierra 10.13.6
      CPU: (4) x64 Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
      Memory: 34.89 MB / 16.00 GB
      Shell: 5.6.2 - /usr/local/bin/zsh
    Binaries:
      Node: 11.0.0 - /usr/local/bin/node
      Yarn: 1.12.1 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 11.4, macOS 10.13, tvOS 11.4, watchOS 4.3
    IDEs:
      Android Studio: 3.2 AI-181.5540.7.32.5056338
      Xcode: 9.4.1/9F2000 - /usr/bin/xcodebuild
    npmPackages:
      react: ^16.6.1 => 16.6.1
      react-native: ^0.57.4 => 0.57.4
    npmGlobalPackages:
      create-react-native-app: 1.0.0
      react-native-cli: 2.0.1
```

## Description

After I enable Hot Reloading and make some change, the app crashes with EXC_BAD_ACCESS. This happens 100% of the time. It doesn't matter what I change, if I just save a file without making any changes the app crashes. I'm running it from Xcode.

This started happening after I upgraded from React Native 0.55.4 to 0.57.4 (latest version).

Stack trace:

```
thread #9, name = 'com.facebook.react.JavaScript', stop reason = EXC_BAD_ACCESS (code=1, address=0x1721a8a48)
    frame #0: 0x000000010b6d19c8 JavaScriptCore`JSC::JSObject::toPrimitive(JSC::ExecState*, JSC::PreferredPrimitiveType) const + 2648
    frame #1: 0x000000010b677e5a JavaScriptCore`JSC::JSValue::toStringSlowCase(JSC::ExecState*, bool) const + 1290
    frame #2: 0x000000010b678322 JavaScriptCore`JSC::JSValue::toWTFStringSlowCase(JSC::ExecState*) const + 882
    frame #3: 0x000000010af331ed JavaScriptCore`JSValueToStringCopy + 157
  * frame #4: 0x0000000106659230 MyApp`facebook::react::Value::toString(this=0x000070000bf7b000) const at Value.cpp:167
    frame #5: 0x00000001066fcd38 MyApp`facebook::react::nativeInjectHMRUpdate(ctx=0x000070000bf7b180, function=0x000000012a65c100, thisObject=0x000000012a6940a0, argumentCount=1, arguments=0x000070000bf7b0b0, exception=0x000070000bf7b138) at JSCExecutor.cpp:116
    frame #6: 0x000000010af21b9f JavaScriptCore`long long JSC::APICallbackFunction::call<JSC::JSCallbackFunction>(JSC::ExecState*) + 495
    frame #7: 0x000004204760902d
    frame #8: 0x000000010aef633d JavaScriptCore`llint_entry + 26817
    frame #9: 0x000004204771f9e3
    frame #10: 0x000000010aef633d JavaScriptCore`llint_entry + 26817
    frame #11: 0x000000010aef633d JavaScriptCore`llint_entry + 26817
    frame #12: 0x000000010aef633d JavaScriptCore`llint_entry + 26817
    frame #13: 0x000000010aef633d JavaScriptCore`llint_entry + 26817
    frame #14: 0x0000042047a2cd46
    frame #15: 0x000000010aef62e2 JavaScriptCore`llint_entry + 26726
    frame #16: 0x00000420479e2517
    frame #17: 0x00000420479df858
    frame #18: 0x00000420479399e2
    frame #19: 0x00000420476de543
    frame #20: 0x00000420479592f8
    frame #21: 0x000000010aeef8fe JavaScriptCore`vmEntryToJavaScript + 244
    frame #22: 0x000000010b3eb846 JavaScriptCore`JSC::Interpreter::executeCall(JSC::ExecState*, JSC::JSObject*, JSC::CallType, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&) + 470
    frame #23: 0x000000010b5ec85e JavaScriptCore`JSC::call(JSC::ExecState*, JSC::JSValue, JSC::CallType, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&) + 62
    frame #24: 0x000000010b6752c9 JavaScriptCore`JSC::boundThisNoArgsFunctionCall(JSC::ExecState*) + 377
    frame #25: 0x000000010aeefa25 JavaScriptCore`vmEntryToNative + 250
    frame #26: 0x000000010b3eb88e JavaScriptCore`JSC::Interpreter::executeCall(JSC::ExecState*, JSC::JSObject*, JSC::CallType, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&) + 542
    frame #27: 0x000000010b5ec9b4 JavaScriptCore`JSC::profiledCall(JSC::ExecState*, JSC::ProfilingReason, JSC::JSValue, JSC::CallType, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&) + 196
    frame #28: 0x000000010af2f2d4 JavaScriptCore`JSObjectCallAsFunction + 468
    frame #29: 0x000000010665a8d1 MyApp`facebook::react::Object::callAsFunction(this=0x00007f9bf0c1a918, thisObj=0x0000000000000000, nArgs=3, args=0x000070000bf7c350) const at Value.cpp:228
    frame #30: 0x000000010665a846 MyApp`facebook::react::Object::callAsFunction(this=0x00007f9bf0c1a918, args=([0] = 0x000000012f3939e0, [1] = 0x000000012f3939c0, [2] = 0x000000012eea5c30)) const at Value.cpp:211
    frame #31: 0x0000000106700706 MyApp`facebook::react::JSCExecutor::callFunction(this=0x000070000bf7c390)::$_3::operator()() const at JSCExecutor.cpp:612
    frame #32: 0x0000000106700423 MyApp`facebook::react::JSCExecutor::callFunction(this=0x00007f9bf0c1a810, moduleId="RCTDeviceEventEmitter", methodId="emit", arguments=0x0000600001e01b40) at JSCExecutor.cpp:606
    frame #33: 0x0000000106722745 MyApp`facebook::react::NativeToJsBridge::callFunction(this=0x0000600001e01b08, executor=0x00007f9bf0c1a810)::$_1::operator()(facebook::react::JSExecutor*) const at NativeToJsBridge.cpp:168
    frame #34: 0x0000000106721cc0 MyApp`void std::__1::__invoke_void_return_wrapper<void>::__call<facebook::react::NativeToJsBridge::callFunction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, folly::dynamic&&)::$_1&, facebook::react::JSExecutor*>(facebook::react::NativeToJsBridge::callFunction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, folly::dynamic&&)::$_1&&&, facebook::react::JSExecutor*&&) [inlined] decltype(__f=0x0000600001e01b08, __args=0x000070000bf7c918)::$_1&>(fp)(std::__1::forward<facebook::react::JSExecutor*>(fp0))) std::__1::__invoke<facebook::react::NativeToJsBridge::callFunction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, folly::dynamic&&)::$_1&, facebook::react::JSExecutor*>(facebook::react::NativeToJsBridge::callFunction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, folly::dynamic&&)::$_1&&&, facebook::react::JSExecutor*&&) at type_traits:4428
    frame #35: 0x0000000106721ca0 MyApp`void std::__1::__invoke_void_return_wrapper<void>::__call<facebook::react::NativeToJsBridge::callFunction(__args=0x0000600001e01b08, __args=0x000070000bf7c918)::$_1&, facebook::react::JSExecutor*>(facebook::react::NativeToJsBridge::callFunction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, folly::dynamic&&)::$_1&&&, facebook::react::JSExecutor*&&) at __functional_base:349
    frame #36: 0x0000000106721829 MyApp`std::__1::__function::__func<facebook::react::NativeToJsBridge::callFunction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, folly::dynamic&&)::$_1, std::__1::allocator<facebook::react::NativeToJsBridge::callFunction(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&&, folly::dynamic&&)::$_1>, void (facebook::react::JSExecutor*)>::operator(this=0x0000600001e01b00, __arg=0x000070000bf7c918)(facebook::react::JSExecutor*&&) at functional:1562
    frame #37: 0x000000010672c554 MyApp`std::__1::function<void (facebook::react::JSExecutor*)>::operator(this=0x0000600001e01b00, __arg=0x00007f9bf0c1a810)(facebook::react::JSExecutor*) const at functional:1913
    frame #38: 0x000000010672c4be MyApp`facebook::react::NativeToJsBridge::runOnExecutorQueue(this=0x0000600001424c70)>)::$_7::operator()() const at NativeToJsBridge.cpp:255
    frame #39: 0x000000010672c43d MyApp`void std::__1::__invoke_void_return_wrapper<void>::__call<facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function<void (facebook::react::JSExecutor*)>)::$_7&>(facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function<void (facebook::react::JSExecutor*)>)::$_7&&&) [inlined] decltype(__f=0x0000600001424c70)>)::$_7&>(fp)(std::__1::forward<>(fp0))) std::__1::__invoke<facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function<void (facebook::react::JSExecutor*)>)::$_7&>(facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function<void (facebook::react::JSExecutor*)>)::$_7&&&) at type_traits:4428
    frame #40: 0x000000010672c42c MyApp`void std::__1::__invoke_void_return_wrapper<void>::__call<facebook::react::NativeToJsBridge::runOnExecutorQueue(__args=0x0000600001424c70)>)::$_7&>(facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function<void (facebook::react::JSExecutor*)>)::$_7&&&) at __functional_base:349
    frame #41: 0x000000010672bff9 MyApp`std::__1::__function::__func<facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function<void (facebook::react::JSExecutor*)>)::$_7, std::__1::allocator<facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function<void (facebook::react::JSExecutor*)>)::$_7>, void ()>::operator(this=0x0000600001424c60)() at functional:1562
    frame #42: 0x0000000106546c86 MyApp`std::__1::function<void ()>::operator(this=0x0000600001424c60)() const at functional:1913
    frame #43: 0x0000000106546988 MyApp`facebook::react::tryAndReturnError(func=0x0000600001424c60)> const&) at RCTCxxUtils.mm:95
    frame #44: 0x000000010652e681 MyApp`facebook::react::RCTMessageThread::tryFunc(this=0x0000600002a1d118, func=0x0000600001424c60)> const&) at RCTMessageThread.mm:60
    frame #45: 0x00000001065322c7 MyApp`facebook::react::RCTMessageThread::runOnQueue(this=0x00006000013a9ec8)>&&)::$_1::operator()() const at RCTMessageThread.mm:73
    frame #46: 0x00000001065321fd MyApp`void std::__1::__invoke_void_return_wrapper<void>::__call<facebook::react::RCTMessageThread::runOnQueue(std::__1::function<void ()>&&)::$_1&>(facebook::react::RCTMessageThread::runOnQueue(std::__1::function<void ()>&&)::$_1&&&) [inlined] decltype(__f=0x00006000013a9ec8)>&&)::$_1&>(fp)(std::__1::forward<>(fp0))) std::__1::__invoke<facebook::react::RCTMessageThread::runOnQueue(std::__1::function<void ()>&&)::$_1&>(facebook::react::RCTMessageThread::runOnQueue(std::__1::function<void ()>&&)::$_1&&&) at type_traits:4428
    frame #47: 0x00000001065321ec MyApp`void std::__1::__invoke_void_return_wrapper<void>::__call<facebook::react::RCTMessageThread::runOnQueue(__args=0x00006000013a9ec8)>&&)::$_1&>(facebook::react::RCTMessageThread::runOnQueue(std::__1::function<void ()>&&)::$_1&&&) at __functional_base:349
    frame #48: 0x0000000106531f09 MyApp`std::__1::__function::__func<facebook::react::RCTMessageThread::runOnQueue(std::__1::function<void ()>&&)::$_1, std::__1::allocator<facebook::react::RCTMessageThread::runOnQueue(std::__1::function<void ()>&&)::$_1>, void ()>::operator(this=0x00006000013a9ec0)() at functional:1562
    frame #49: 0x0000000106546c86 MyApp`std::__1::function<void ()>::operator(this=0x00006000013a9ec0)() const at functional:1913
    frame #50: 0x000000010652e429 MyApp`::___ZN8facebook5react16RCTMessageThread8runAsyncENSt3__18functionIFvvEEE_block_invoke(.block_descriptor=0x00006000013a9ea0) at RCTMessageThread.mm:40
    frame #51: 0x000000010959a62c CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
    frame #52: 0x0000000109599de0 CoreFoundation`__CFRunLoopDoBlocks + 336
    frame #53: 0x0000000109594c2b CoreFoundation`__CFRunLoopRun + 2779
    frame #54: 0x0000000109593e11 CoreFoundation`CFRunLoopRunSpecific + 625
    frame #55: 0x00000001064cdb4f MyApp`::+[RCTCxxBridge runRunLoop](self=RCTCxxBridge, _cmd="runRunLoop") at RCTCxxBridge.mm:252
    frame #56: 0x000000010724a732 Foundation`__NSThread__start__ + 1221
    frame #57: 0x000000010d719661 libsystem_pthread.dylib`_pthread_body + 340
    frame #58: 0x000000010d71950d libsystem_pthread.dylib`_pthread_start + 377
    frame #59: 0x000000010d718bf9 libsystem_pthread.dylib`thread_start + 13
```

## Reproducible Demo

I can't provide a reproducible demo at the moment.

**Comment by sryze** [comment]
After downgrading 0.57.2 hot reloading works again.

**Comment by sryze** [comment]
I think it's the same issue as #22116, feel free to close if it is

**Comment by Titozzz** [comment]
Closed as duplicate from #22116

**Comment by kk412027247** [comment]
After turn on the **Debug JS Remotely**, there is no more crash.
